examples/04_scalability.py

```diff
@@ -0,0 +1,179 @@
+from dirty_cat import SimilarityEncoder
+
+print('\nBasic similarity encoding')
+
+sim_enc = SimilarityEncoder(similarity='ngram', handle_unknown='ignore')
+
+y = df['Violation Type']
+
+# clean columns
+transformers = [('one_hot', OneHotEncoder(sparse=False, handle_unknown='ignore'),
+                 ['Alcohol',
+                  'Arrest Type',
+                  'Belts',
+                  'Commercial License',
+                  'Commercial Vehicle',
+                  'Fatal',
+                  'Gender',
+                  'HAZMAT',
+                  'Property Damage',
+                  'Race',
+                  'Work Zone']),
+                ('pass', 'passthrough', ['Year']),
+                ]
+
+column_trans = ColumnTransformer(
+    # adding the dirty column
+    transformers=transformers + [('sim_enc', sim_enc, ['Description'])],
+    remainder='drop')
+
+
+t0 = time()
+X = column_trans.fit_transform(df)
+t1 = time()
+print('Time to vectorize: %s' % (t1 - t0))
+################################################################################
+# We can run a cross-validation
+from sklearn import linear_model, pipeline, model_selection
+log_ref = linear_model.LogisticRegression()
+
+model = pipeline.make_pipeline(column_trans, log_ref)
+print("Cross-validation score: %s" %
+      resource_used(model_selection.cross_val_score)(model, df, y))
+
+
+################################################################################
+# SimilarityEncoder with most frequent strategy
+# ---------------------------------------------
+#
+# The most frequent strategy selects the n most frequent values in a dirty
+# categorical variable to reduce the dimensionality of the problem and thus
+# speed things up. We select manually the number of prototypes we want to use.
+
+print('\nSimilarity encoding with a most frequent strategy')
+
+sim_enc = SimilarityEncoder(similarity='ngram', categories='most_frequent', n_prototypes=100)
+
+column_trans = ColumnTransformer(
+    # adding the dirty column
+    transformers=transformers + [('sim_enc', sim_enc, ['Description'])],
+    remainder='drop')
+
+from time import time
+
+t0 = time()
+X = column_trans.fit_transform(df)
+t1 = time()
+print('Time to vectorize: %s' % (t1 - t0))
+
+################################################################################
+# Check now that prediction is still as good
+model = pipeline.make_pipeline(column_trans, log_ref)
+print("Cross-validation score: %s" %
+      resource_used(model_selection.cross_val_score)(model, df, y))
+
+
+################################################################################
+# SimilarityEncoder with k-means strategy
+# ---------------------------------------
+#
+# K-means strategy is also a dimensionality reduction technique. But we apply
+# a K-means and nearest neighbors algorithm to find the prototypes. The number
+# of prototypes is set manually.
+
+print('\nSimilarity encoding with a k-means strategy')
```

Comment:
We should probably remove this print: on the webpage it's not useful given the section title.